es-core/src/Window.cpp

```diff
@@ -335,8 +335,8 @@ void Window::setHelpPrompts(const std::vector<HelpPrompt>& prompts, const HelpSt
 				// yes, it has!
 
 				// can we combine? (dpad only)
-				if((it->first == "up/down" && addPrompts.at(mappedTo->second).first == "left/right") ||
-					(it->first == "left/right" && addPrompts.at(mappedTo->second).first == "up/down"))
+				if((strcmp(it->first, "up/down") == 0 && addPrompts.at(mappedTo->second).first == "left/right") ||
```

Comment:
I was hoping @nschloe would answer first :D I've learned time and time again not to question C, as I am never better off after it. Just the past hour I've been trying to understand why a simple division assignment to an unsigned char wasn't working as intended, in EmulationStation... 

So yes, in principle your assessment would be correct. I wonder if the compiler does some sort of smart optimization or conversion to make it work - or, if as you say, it actually never really worked properly. 

Then again, maybe I am happier not knowing the answer :)